aws/terraform/infra/s3.tf

```diff
@@ -0,0 +1,6 @@
+resource "aws_s3_bucket" "data_sets" {
+  bucket = local.datasets_s3_bucket_name
```

Comment:
It looks like here is a high chance of bucket name collision. For example, I've just applied this and the bucket is named `metaflow-s3-prod-us-west-2`. Next person who tries to apply this will run into trouble as bucket names are globally unique. 

I wonder if it should generate a random name? Or maybe let's at least put a big warning in the README that `var.app` should be set to something really unique by the user before trying to apply this.

Reply:
This is a great idea, yea I think we should try to make this Terraform code as easy to run for someone coming in fresh .

That being said, I think something as long lasting and important as a `s3` bucket name should be named by the end user. I'd lean towards your second idea, where this is inputted by the user. 

I can make the change and push it to this branch, just let me know that you agree

Reply:
This is a great idea, yea I think we should try to make this Terraform code as easy to run for someone coming in fresh .

That being said, I think something as long lasting and important as a `s3` bucket name should be named by the end user. I'd lean towards your second idea, where this is inputted by the user. 

I can make the change and push it to this branch, just let me know that you agree

Reply:
Makes sense!

Reply:
Resolved in my latest commit!